domain-mapping/inc/sunrise.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+defined( 'ABSPATH' ) or die();
+
+/**
+ * If the COOKIE_DOMAIN is set, then we cannot proceed as the login will not be
+ * able to function with the mapped domain.
+ */
+if ( defined( 'COOKIE_DOMAIN' ) ) {
+    wp_die( __( "Dark Matter's single sign on will not work if a COOKIE_DOMAIN is defined.", 'dark-matter' ) );
+}
+
+wp_cache_add_global_groups( 'dark-matter' );
+
+if ( false === defined( 'SUNRISE_LOADED' ) ) {
+    define( 'SUNRISE_LOADED', true );
+}
+
+/**
+ * Cannot utilise plugin_dir_path() as the inner function used is not available
+ * and this is preferable to include more files than is realistically needed.
+ */
+$dirname = str_replace( '/inc', '', dirname( __FILE__ ) );
+
+/**
+ * Load the necessary parts of Dark Matter in to place.
+ */
+require_once $dirname . '/classes/DM_Domain.php';
+require_once $dirname . '/api/DarkMatter_Domains.php';
+require_once $dirname . '/api/DarkMatter_Primary.php';
+
+/**
+ * Attempt to find the Site.
+ */
+$fqdn = $_SERVER['HTTP_HOST'];
+
+global $dm_domain;
+$dm_domain = DarkMatter_Domains::instance()->get( $fqdn );
+
+if ( $dm_domain && $dm_domain->active ) {
+    /**
+     * Prepare all the global variables. This is require irrespective of whether
+     * it is a primary or secondary domain.
+     */
+    global $current_blog, $original_blog;
+    $current_blog  = get_site( $dm_domain->blog_id );
+
+    global $current_site;
+    $current_site = WP_Network::get_instance( $current_blog->site_id );
+
+    global $blog_id; $blog_id = $current_blog->blog_id;
+    global $site_id; $site_id = $current_blog->site_id;
+
+    /**
+     * Dark Matter will disengage if the website is no longer public or is
+     * archived or deleted.
+     */
+    if ( (int) $current_blog->public > 0 && $current_blog->archived !== '0' && $current_blog->deleted !== '0' ) {
```

Comment:
>- Choosing "deactive" marks it as "deleted" in the database. This causes it not to show up on the blog. If you visit the blog, a message appears which says "This user has elected to delete their account and the content is no longer available." This seems like a somewhat inaccurate message.
>- Choosing "archive" marks it as "archived" in the database. If you visit the blog, a message appears which says "This blog has been archived or suspended."
>- Choosing "delete" actually deletes the blog, and all its posts and settings from the database.

Based on the above understanding, on platforms where the host is not the client, the host might want to archive the site but have it still mapped so that the message can be shown that the blog is suspended.